test/storage/index_populator_test.cpp

```diff
@@ -0,0 +1,177 @@
+
+#include "storage/index/index_populator.h"
+#include <string>
+#include <vector>
+#include "catalog/catalog.h"
+#include "catalog/catalog_sql_table.h"
+#include "storage/index/index_builder.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier::storage::index {
+struct IndexPopulatorTest : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+
+    catalog_ = new catalog::Catalog(txn_manager_);
```

Comment:
[Major] The test probably can not be built because the Catalog does not have such constructor. But I think it's totally OK because it just needs some efforts to fix.